controllers/controller.js

```diff
@@ -11,41 +11,42 @@ mongoose.connect('mongodb://test:test@ds257627.mlab.com:57627/first-blog-mern',
 
 const dataService = new service();
 module.exports = app => {
-
-//    const homeFunc = (req, res) => {
-//        Post.find({}, function(err, data){
-//            if (err) throw (err);
-//            res.render('posts', {posts: data});
-//        });
-//    };
     
-    const homeFunc = (req, res) => {
-        const data = dataService.getAllPosts();
-        console.log(data);
-        res.render('posts', {posts: data});
-    };
 
-    app.get('/', homeFunc);
 
+    app.get(['/', '/posts'], function(req, res) {
+        dataService.getAllPosts().then(posts => {
+            console.log(posts);
+            res.render('posts', {posts: posts});
+        }).catch({
+            res.status(500);
+        });
+    });
     
-    app.get('/posts', homeFunc);
-
     app.get('/posts/:post_id', function(req, res){
-        Post.find({_id: req.params.post_id}, function(err, data){
-            if (err) throw (err);
-            res.render('post_detail', {post: data});
+        dataService.getOnePost(req).then((post) => {
+            res.render('post_detail', {post: post});
         });
     });
 
     app.get('/post_new', function(req, res){
         res.render('post_new');
     });
 
+    app.post('/post_new', urlencodedParser, function(req, res){
+        // get data from view and add to mongodb
+        const newPost = dataService.postNewPost(req).then((post) => {
```

Comment:
What is the `newPost` variable used for?

Reply:
fixed